pkg/util/template.go

```diff
@@ -214,6 +220,38 @@ func configMapEntriesGenerator(
 	}
 }
 
+// lookup does a dot-separated path lookup on the input struct or map. If the input or any of
+// its children on the targeted path are not a map or struct, it returns nil.
+func lookup(path string, input interface{}) interface{} {
+	obj := reflect.ValueOf(input)
+
+	components := strings.Split(path, ".")
+
+	for i := 0; i < len(components); {
+		component := components[i]
+
+		switch obj.Kind() {
```

Comment:
It looks like Sprig made the opinionated choice to use `map[string]interface{}` only https://github.com/Masterminds/sprig/blob/39e4d5d0e0d566a256746e59749821155f209d11/dict.go#L8

Do you know if we may ever run into cases where a struct is passed to this function?

Reply:
Map only is fine. I was trying to be super-flexible but probably unnecessary :-).

Reply:
Fixed.